dds/idl/marshal_generator.cpp

```diff
@@ -615,81 +671,174 @@ namespace {
       be_global->impl_ <<
         "  const Encoding& encoding = strm.encoding();\n"
         "  ACE_UNUSED_ARG(encoding);\n";
-      std::vector<string> code;
-      code.push_back("if (!strm.read_delimiter(total_size)) {");
-      code.push_back("  return false;");
-      code.push_back("}");
+      const string code =
+        "    if (!strm.read_delimiter(total_size)) {\n"
+        "      return false;\n"
+        "    }\n";
       generate_dheader_code(code, !primitive);
```

Comment:
```suggestion
      generate_dheader_code(
        "    if (!strm.read_delimiter(total_size)) {\n"
        "      return false;\n"
        "    }\n", !primitive);
```

And in the other places.